### PR TITLE
Disable package controller based on cluster specification

### PR DESCRIPTION
**Pull request by TerryHowe**

### CODE REVIEW

File: pkg/curatedpackages/packageinstaller_test.go [hunk 1]
```diff
@@ -93,3 +93,40 @@ func TestPackageInstallerFailWhenPackageFails(t *testing.T) {
...
+
+	tt.packageClient.EXPECT().CreatePackages(tt.ctx, tt.packagePath, tt.kubeConfigPath).Return(nil)
+
+	tt.command.InstallCuratedPackages(tt.ctx)
+}
+
+func TestIsPackageControllerDisabled(t *testing.T) {
+	tt := newPackageInstallerTest(t)
+
+	if curatedpackages.IsPackageControllerDisabled(nil) == true {
```
Comment (vincentni) [inline pkg/curatedpackages/packageinstaller_test.go:111]: It is unnecessary to do equality check on booleans.

File: pkg/curatedpackages/packageinstaller_test.go [hunk 2]
```diff
@@ -93,3 +93,40 @@ func TestPackageInstallerFailWhenPackageFails(t *testing.T) {
...
+		t.Errorf("nil cluster should be enabled")
+	}
+
+	if curatedpackages.IsPackageControllerDisabled(tt.spec.Cluster) == true {
```
Comment (vincentni) [inline pkg/curatedpackages/packageinstaller_test.go:115]: same comment here

File: pkg/curatedpackages/packageinstaller_test.go [hunk 3]
```diff
@@ -93,3 +93,40 @@ func TestPackageInstallerFailWhenPackageFails(t *testing.T) {
...
+		t.Errorf("nil package controller should be enabled")
+	}
+
+	tt.spec.Cluster.Spec.Packages = &anywherev1.PackageConfiguration{
+		Disable: false,
+	}
+	if curatedpackages.IsPackageControllerDisabled(tt.spec.Cluster) == true {
```
Comment (vincentni) [inline pkg/curatedpackages/packageinstaller_test.go:122]: same comment here

File: pkg/curatedpackages/packageinstaller_test.go [hunk 4]
```diff
@@ -93,3 +93,40 @@ func TestPackageInstallerFailWhenPackageFails(t *testing.T) {
...
+		t.Errorf("package controller should be enabled")
+	}
+
+	tt.spec.Cluster.Spec.Packages = &anywherev1.PackageConfiguration{
+		Disable: true,
+	}
+	if curatedpackages.IsPackageControllerDisabled(tt.spec.Cluster) == false {
```
Comment (vincentni) [inline pkg/curatedpackages/packageinstaller_test.go:129]: same comment here

File: pkg/curatedpackages/packageinstaller.go [hunk 5]
```diff
@@ -25,6 +26,18 @@ type Installer struct {
 	mgmtKubeconfig    string
 }
 
+// IsPackageControllerDisabled detect if the package controller is disabled.
+func IsPackageControllerDisabled(cluster *anywherev1.Cluster) bool {
+	if cluster != nil {
```
Comment (a-cool-train) [inline pkg/curatedpackages/packageinstaller.go:31]: Maybe something like return cluster != nil && cluster.Spec.Packages != nil && cluster.Spec.Packages.Disable?

File: pkg/curatedpackages/packageinstaller_test.go (hunk 1 above)
Comment (TerryHowe) [reply pkg/curatedpackages/packageinstaller_test.go]: Anyway, fixed

**Review by vincentni — APPROVED** [review]
/lgtm
/approve

**Pull request merged**